### PR TITLE
dict update syntax improvement to list_models()

### PR DESCRIPTION
**Pull request by lstein**
This is a code cleanup enhancement originally contributed by @devops117 which didn't get into the 2.2 release. Reissuing the PR now in order to base against main.

### CODE REVIEW

File: ldm/invoke/model_cache.py [hunk 1]
```diff
@@ -145,10 +145,14 @@ def list_models(self) -> dict:
             else:
                 status = 'not loaded'
 
-            models[name]={
-                'status' : status,
-                'description' : description
-            }
+            models.update(
+                {
+                    name: {
```
Comment (psychedelicious) [inline ldm/invoke/model_cache.py:150]: The syntax is a new dict that overwrites the existing keys, so this needs quotes.

```suggestion
                    'name': {
```

**Review by psychedelicious — CHANGES_REQUESTED** [review]

File: ldm/invoke/model_cache.py (hunk 1 above)
Comment (mebelz) [reply ldm/invoke/model_cache.py:150]: I personally don't understand the original change, or psychedelicious's suggestion.
In my opinion, the original change is switching between equivalent ways of updating a dict (the pre-commit being more readable than the commit).
psychedelicious's suggestion changes the code such that the keys to models are only 'name', and its constantly overwritten.
Anyway I suggest adding a code comment explaining what is the exact merit of the change, for future reference.

Comment (tildebyte) [reply ldm/invoke/model_cache.py:150]: @psychedelicious is correct; this code will throw an unknown key exception without the quotes.

Having said that, they're "a horse apiece" - using Dict's `update()` method is more Pythonic, but I find them equally readable, and equally obvious.

Comment (psychedelicious) [reply ldm/invoke/model_cache.py:150]: Actually @tildebyte I think @mebelz is correct, `name` is a variable in this context - the name of the model to be updated.

**Comment by lstein** [comment]
Some of the CI tests are stuck and it isn't worth it to unstick them for this bit of syntactic sugar. So closing.

**Pull request closed without merging**